src/convertToObject.js

```diff
@@ -6,7 +6,19 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const arrayOfSource = sourceString
+    .split(';')
+    .filter((part) => part.trim() !== '');
+  const MappedArrayOfSource = arrayOfSource.map((part) => {
+    return part.split(':');
+  });
```

Comment:
Instead of mapping and then using forEach, you can reduce the array directly to an object to make the code more concise and avoid the extra iteration.